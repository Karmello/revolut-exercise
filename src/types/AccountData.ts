type AccountData = {
  currency: string
  amount: string
}

export default AccountData
