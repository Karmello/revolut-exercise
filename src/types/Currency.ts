enum Currency {
  USD = 'USD',
  EUR = 'EUR',
  GBP = 'GBP',
}

export default Currency
