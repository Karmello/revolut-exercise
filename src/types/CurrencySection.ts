enum CurrencySection {
  Base = 'BASE',
  Target = 'TARGET',
}

export default CurrencySection
