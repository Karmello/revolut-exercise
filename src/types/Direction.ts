enum Direction {
  Prev = 'PREV',
  Next = 'NEXT',
}

export default Direction
