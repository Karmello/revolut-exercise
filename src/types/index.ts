export { default as AccountData } from './AccountData'
export { default as Currency } from './Currency'
export { default as CurrencySection } from './CurrencySection'
export { default as Direction } from './Direction'
export { default as RatesData } from './RatesData'
