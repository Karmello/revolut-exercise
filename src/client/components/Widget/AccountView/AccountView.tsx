import React from 'react'
import { hot } from 'react-hot-loader'
import { Header } from 'semantic-ui-react'

import Currency from 'types/Currency'

type Props = {
  currency: Currency
}

const AccountView = ({ currency }: Props) => (
  <>
    <Header as="h1" content={currency} />
  </>
)

export default hot(module)(AccountView)
