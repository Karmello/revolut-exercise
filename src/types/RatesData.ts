import Currency from './Currency'

type RatesData = {
  base: Currency
  rates: {
    [key: string]: number
  }
}

export default RatesData
