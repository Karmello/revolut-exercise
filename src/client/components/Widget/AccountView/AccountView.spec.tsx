import React from 'react'
import { shallow } from 'enzyme'

import { CurrencySection, Currency } from 'types'
import AccountView from './AccountView'

describe('AccountView component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <AccountView
        type={CurrencySection.Base}
        currency={Currency.EUR}
        amount={100}
        exchangeInfo="€1.00 = $1.23"
        formValue="23.45"
        makingExchangeRequest={false}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should match snapshot', () => {
    const wrapper = shallow(
      <AccountView
        type={CurrencySection.Target}
        currency={Currency.GBP}
        amount={45}
        exchangeInfo="€1.00 = $1.23"
        formValue="14.10"
        makingExchangeRequest={true}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
