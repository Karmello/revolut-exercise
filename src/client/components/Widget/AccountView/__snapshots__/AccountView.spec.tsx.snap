// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountView component should match snapshot 1`] = `ShallowWrapper {}`;

exports[`AccountView component should match snapshot 2`] = `ShallowWrapper {}`;
